pkg/distributor/ha_tracker.go

```diff
@@ -220,16 +235,26 @@ func (c *haTracker) checkReplica(ctx context.Context, userID, cluster, replica s
 	}
 	key := fmt.Sprintf("%s/%s", userID, cluster)
 	now := mtime.Now()
+
 	c.electedLock.RLock()
 	entry, ok := c.elected[key]
+	clusters := c.clusters[userID]
 	c.electedLock.RUnlock()
+
 	if ok && now.Sub(timestamp.Time(entry.ReceivedAt)) < c.cfg.UpdateTimeout+c.updateTimeoutJitter {
 		if entry.Replica != replica {
 			return replicasNotMatchError(replica, entry.Replica)
 		}
 		return nil
 	}
 
+	if !ok {
+		// If we don't know about this cluster yet and we have reached the limit for number of clusters, we error out now.
+		if limit := c.limits.MaxHAClusters(userID); limit > 0 && clusters+1 > limit {
```

Comment:
These samples are discarded. We need to increase `validation.DiscardedSamples` with a new `reason`.